spec/snowflake/client_spec.rb

```diff
@@ -0,0 +1,146 @@
+require "spec_helper"
+
+RSpec.describe Snowflake::Client do
+  subject(:client) { described_class.new }
+
+  describe "#connect" do
+    context "when the account is empty" do
+      it "will return an error" do
+        expect {
+          client.connect("","","","","","","")
+        }.to raise_error(ArgumentError, "Snowflake Config Creation Error: '260000: account is empty'")
+      end
+    end
+
+    context "when all values are valid" do
+      it "will not raise an error" do
+        expect(
+          client.connect("acc","warehouse","database","schema","user","pwd","role")
+        ).to eq(false)
+      end
+    end
+
+    context "with values from env" do
+      it "will not raise an error" do
+        expect(
+          client.connect(
+            ENV["SNOWFLAKE_ACCOUNT"],
+            ENV["SNOWFLAKE_WAREHOUSE"],
+            "",
+            "",
+            ENV["SNOWFLAKE_USER"],
+            ENV["SNOWFLAKE_PASSWORD"],
+            ""
+          )
+        ).to eq(false)
+      end
+    end
+  end
+
+  describe "#fetch" do
+    before do
+      client.connect(
+        ENV["SNOWFLAKE_ACCOUNT"],
+        ENV["SNOWFLAKE_WAREHOUSE"],
+        #"ruby_snowflake_client_testing",
+        "",
+        "",
+        ENV["SNOWFLAKE_USER"],
+        ENV["SNOWFLAKE_PASSWORD"],
+        ""
+      )
+    end
+
+    let(:query) { "" }
+    let(:result) { client.fetch(query) }
+
+    context "with a simple query returning string" do
+      let(:query) { "SELECT 1;" }
+      it "should return a Snowflake::Result" do
```

Comment:
nitpick: consistent spacing
```suggestion

      it "should return a Snowflake::Result" do
```